app/controllers/circle_check_scores_controller.rb

```diff
@@ -2,9 +2,15 @@ class CircleCheckScoresController < ApplicationController
   before_action :find_score, only: :update
 
   def create
-    score = CircleCheckScore.create! score_params
-    redirect_to circle_check_scores_path, notice: 'Score was saved.'
-    PrivatePub.publish_to '/scoreboard', score
+    deny_access && return unless current_user.has_role? :circle_check_scorer
+    score = CircleCheckScore.new score_params
+    if score.save
+      redirect_to circle_check_scores_path, notice: 'Score was saved.'
+      PrivatePub.publish_to '/scoreboard', score
+    else
+      flash[:error] = 'Invalid Score'
```

Comment:
Take a look at what some other controllers do - there's a way to dynamically return even more information that this.